packages/minimongo/minimongo.js

```diff
@@ -221,14 +221,26 @@ LocalCollection.Cursor.prototype.fetch = function () {
  * @locus Anywhere
  * @returns {Number}
  */
-LocalCollection.Cursor.prototype.count = function () {
+LocalCollection.Cursor.prototype.count = function (applySkipLimit) {
   var self = this;
+  // Have to respect `applySkipLimit`, so we remove limit & skip if `applySkipLimit`
```

Comment:
This doesn't look right to me. At the very least if you're doing something like this you need to put the code that reverts skip and limit at the end in a `finally` block, in case `self._getRawObjects` throws.

But since `_getRawObjects` yields, this has the consequence that if you are waiting on the result of `foo.count({applySkipLimit: true})` and then in another fiber you call `foo.find()` you're going to get the full results of the query without skip and limit.

A good approach instead would be to somehow clone the instance to a new one and remove `skip` and `limit` there.

Hmm, I guess I'm actually confused about this code altogether -- if you don't pass `applySkipLimit` it just ignores the values of `skip` and `limit`? So why do we need to remove the values on `self` here?